CMakeLists.txt

```diff
@@ -191,7 +193,10 @@ IF (KOKKOS_HAS_TRILINOS)
   FOREACH(CUDAFE_FLAG ${KOKKOS_CUDAFE_OPTIONS})
     SET(KOKKOSCORE_CUDAFE_OPTIONS "${KOKKOSCORE_CUDAFE_OPTIONS} -Xcudafe ${CUDAFE_FLAG}")
   ENDFOREACH()
-  SET(KOKKOSCORE_CXX_FLAGS "${KOKKOSCORE_COMPILE_OPTIONS} ${CMAKE_CXX${KOKKOS_CXX_STANDARD}_STANDARD_COMPILE_OPTION} ${KOKKOSCORE_CUDA_OPTIONS} ${KOKKOSCORE_CUDAFE_OPTIONS}")
+  FOREACH(XCOMP_FLAG ${KOKKOS_XCOMPILER_OPTIONS})
+    SET(KOKKOSCORE_XCOMPILER_OPTIONS "${KOKKOSCORE_XCOMPILER_OPTIONS} -Xcompiler ${XCOMP_FLAG}")
+  ENDFOREACH()
+  SET(KOKKOSCORE_CXX_FLAGS "${KOKKOSCORE_COMPILE_OPTIONS} ${CMAKE_CXX${KOKKOS_CXX_STANDARD}_STANDARD_COMPILE_OPTION} ${KOKKOSCORE_CUDA_OPTIONS} ${KOKKOSCORE_CUDAFE_OPTIONS} ${KOKKOSCORE_XCOMPILER_OPTIONS}")
```

Comment:
OK. Do I understand correctly that this is the only real change and the rest is not strictly necessary?

Reply:
I don't strictly need to add the function `GLOBAL_RESET`. I can revert this. Other changes to convert things to use `GLOBAL_APPEND` should just be kept since they clean things up.

I initially thought this had to do with funky behavior in `GLOBAL_SET` - but that was not the problem.

Reply:
It's OK. I don't feel strongly about it.